### DISCUSSION TITLE
Network switch handler isn't triggered

### DISCUSSION THREAD

**Discussion by RomualdH** [WalletConnect/walletconnect-monorepo#2610] [Developer Support] [answered]
Hi there!

Wallet and account connections are working fine.
But when trying to switch the network from the Web3Modal networks, the `ethereumClient.watchNetwork` handler isn't triggered. 
However, the network is correctly changed on the wallet.

Reproduction URL: https://app-git-feat-walletconnect2-betswirl.vercel.app

**Comment by xzilja** [comment]
You shouldn't use methods on `ethereumClient`, instead use wagmi actions (or hooks if you use react) for this.

**Reply by RomualdH** [reply]
The issue persists even using `wagmi` actions. You can test on the same repro URL.

The difference of use case with the standard is that the user is redirected to another URL on the `switchNetwork` callback.
Is there a session that I need to clean up/disconnect before redirecting?

Thank you in advance for your help!

**Reply by RomualdH** [reply]
Here's the test:
1) Connect a wallet
2) Click on the network icon (Polygon is the default) and change to another network. Network will change and you'll be redirected (page will refresh).
3) Repeating step 2) do not work. The network is changed on the wallet, but the callback isn't triggered.

**Reply by RomualdH** [reply]
NB: seems the issue is only happening using [Frame](https://frame.sh/) wallet.

**Reply by xzilja** [reply, marked as answer]
They might not be sending correct events in that case.

**Comment by glitch-txs** [comment]
Tested with Trust Wallet and works fine, are you able to reproduce the issue here?
https://lab.web3modal.com/
Maybe you could contact Frame for this issue

**Reply by RomualdH** [reply]
I'm not. It's working on the lab with Frame...

**Reply by glitch-txs** [reply]
I guess the issue might be related to the code or the versioning. Could you provide a minimal reproduction example repo?

**Comment by RomualdH** [comment]
This issue is related to Frame cf https://github.com/floating/frame/issues/1579
